--- a/Atmel-Video-Player.sh
+++ b/Atmel-Video-Player.sh
@@ -10,12 +10,18 @@
 
 	dd if=/dev/zero of=/dev/fb1
 else
+	# find screen geometry
+	resolution=`fbset | sed -n -e "s/^mode \"\([^-]*\).*/\1/p"`
+	V_WIDTH=`echo $resolution | cut -f1 -d 'x'`
+	V_HEIGHT=`echo $resolution | cut -f2 -d 'x'`
+	#echo "w=${V_WIDTH}, h=${V_HEIGHT}"
+
 	# Hack to prevent the pipeline from being stuck
-	(sleep 25 && killall gst-launch-1.0)&
+	(sleep 26 && killall -9 gst-launch-1.0 > /dev/null 2>&1)&
 	if [ -d /proc/asound/card0 ]
 	then
-		gst-launch-1.0 uridecodebin uri=file:///opt/VideoPlayer/media/Tech_on_Tour-Atmel_Visits_Washington_DC.mp4 expose-all-streams=false name=srcVideo caps="video/x-h264\;audio/x-raw" srcVideo. ! h264parse ! queue ! g1h264dec ! video/x-raw,width=800,height=480 ! g1fbdevsink zero-memcpy=true max-lateness=-1 async=false enable-last-sample=false srcVideo. ! queue ! audioconvert ! audio/x-raw,format=S16LE ! alsasink async=false enable-last-sample=false
+		gst-launch-1.0 uridecodebin uri=file:///opt/VideoPlayer/media/Tech_on_Tour-Atmel_Visits_Washington_DC.mp4 expose-all-streams=false name=srcVideo caps="video/x-h264\;audio/x-raw" srcVideo. ! h264parse ! queue ! g1h264dec ! video/x-raw,width=${V_WIDTH},height=${V_HEIGHT} ! g1fbdevsink zero-memcpy=true max-lateness=-1 async=false enable-last-sample=false srcVideo. ! queue ! audioconvert ! audio/x-raw,format=S16LE ! alsasink async=false enable-last-sample=false
 	else
-		gst-launch-1.0 uridecodebin uri=file:///opt/VideoPlayer/media/Tech_on_Tour-Atmel_Visits_Washington_DC.mp4 expose-all-streams=false name=srcVideo caps="video/x-h264" srcVideo. ! h264parse ! queue ! g1h264dec ! video/x-raw,width=800,height=480 ! g1fbdevsink zero-memcpy=true max-lateness=-1 async=false enable-last-sample=false
+		gst-launch-1.0 uridecodebin uri=file:///opt/VideoPlayer/media/Tech_on_Tour-Atmel_Visits_Washington_DC.mp4 expose-all-streams=false name=srcVideo caps="video/x-h264" srcVideo. ! h264parse ! queue ! g1h264dec ! video/x-raw,width=${V_WIDTH},height=${V_HEIGHT} ! g1fbdevsink zero-memcpy=true max-lateness=-1 async=false enable-last-sample=false
 	fi
 fi
