diff --git a/board/Config.in b/board/Config.in
index 68e13d8..7fe7dc2 100644
--- a/board/Config.in
+++ b/board/Config.in
@@ -156,6 +156,19 @@ config	CONFIG_AT91SAM9X5EK
 	help
 	  Use the AT91SAM9X5EK Development board
 
+config	CONFIG_AT91ARIAG25
+	bool "at91ariag25"
+	select CONFIG_DDR2
+	select ALLOW_SDCARD
+	select ALLOW_HSMCI
+	select ALLOW_CPU_CLK_400MHZ
+	select ALLOW_CRYSTAL_12_000MHZ
+	select ALLOW_PIO3
+	select CPU_HAS_PMECC
+	help
+	  Use the ARIAG25 SoM
+
+
 config	CONFIG_AT91SAM9N12EK
 	bool "at91sam9n12ek"
 	select CONFIG_DDR2
@@ -205,6 +218,7 @@ config CONFIG_CHIP
 	default "AT91SAM9G20"	if CONFIG_AT91SAM9G20EK
 	default "AT91SAM9G45"	if CONFIG_AT91SAM9M10G45EK
 	default "AT91SAM9X5"	if CONFIG_AT91SAM9X5EK
+	default "AT91SAM9X5"	if CONFIG_AT91ARIAG25
 	default "AT91SAM9N12"	if CONFIG_AT91SAM9N12EK
 	default "AT91SAMA5D3X"	if CONFIG_AT91SAMA5D3XEK
 	help
@@ -221,6 +235,7 @@ config CONFIG_BOARD
 	default "at91sam9g10ek"	if CONFIG_AT91SAM9G10EK
 	default "at91sam9g20ek"	if CONFIG_AT91SAM9G20EK
 	default "at91sam9x5ek"	if CONFIG_AT91SAM9X5EK
+	default "at91ariag25"	if CONFIG_AT91ARIAG25
 	default "at91sam9n12ek"	if CONFIG_AT91SAM9N12EK
 	default "at91sam9m10g45ek"	if CONFIG_AT91SAM9M10G45EK
 	default "at91sama5d3xek"	if CONFIG_AT91SAMA5D3XEK
@@ -239,6 +254,7 @@ config CONFIG_MACH_TYPE
 	default "0x658"	if CONFIG_AT91SAM9G20EK
 	default "1830"	if CONFIG_AT91SAM9M10G45EK	# 1830
 	default "3373"	if CONFIG_AT91SAM9X5EK		# 3373
+	default "4149"	if CONFIG_AT91ARIAG25
 	default "9999"	if CONFIG_AT91SAM9N12EK
 	default "9999"	if CONFIG_AT91SAMA5D3XEK
 	help
@@ -265,6 +281,7 @@ config CONFIG_TOP_OF_MEMORY
 	default "0x304000"	if CONFIG_AT91SAM9G20EK
 	default "0x30A000"	if CONFIG_AT91SAM9M10G45EK
 	default "0x308000"	if CONFIG_AT91SAM9X5EK
+	default "0x308000"	if CONFIG_AT91ARIAG25
 	default "0x308000"	if CONFIG_AT91SAM9N12EK
 	default "0x310000"	if CONFIG_AT91SAMA5D3XEK
 	help
diff --git a/include/board.h b/include/board.h
index ce75aa1..d30dc82 100644
--- a/include/board.h
+++ b/include/board.h
@@ -64,6 +64,10 @@
 #include "at91sam9x5ek.h"
 #endif
 
+#ifdef CONFIG_AT91ARIAG25
+#include "at91ariag25.h"
+#endif
+
 #ifdef CONFIG_AT91SAM9N12EK
 #include "at91sam9n12ek.h"
 #endif
