diff --git a/board/at91sama5d3xek/board.mk b/board/at91sama5d3xek/board.mk
index 604a024..3b099eb 100644
--- a/board/at91sama5d3xek/board.mk
+++ b/board/at91sama5d3xek/board.mk
@@ -2,18 +2,18 @@ $(shell $(CC) --target-help > tmp.file)
 gcc_cortexa5=$(shell grep cortex-a5 tmp.file)
 
 ifeq (, $(findstring cortex-a5,$(gcc_cortexa5)))
-CPPFLAGS += -DCONFIG_AT91SAMA5D3XEK
+CPPFLAGS += -DCONFIG_AT91SAMA5D3XEK -march=armv7-a
 
 ASFLAGS += \
-	-DCONFIG_AT91SAMA5D3XEK
+	-DCONFIG_AT91SAMA5D3XEK -march=armv7-a
 else
 CPPFLAGS += \
 	-DCONFIG_AT91SAMA5D3XEK \
-	-mtune=cortex-a5
+	-mtune=cortex-a5 -march=armv7-a
 
 ASFLAGS += \
 	-DCONFIG_AT91SAMA5D3XEK
-	-mcpu=cortex-a5
+	-mcpu=cortex-a5 -march=armv7-a
 endif
 
 $(shell rm tmp.file)
