--- a/configure.ac	2014-08-28 02:20:36.000000000 +0200
+++ b/configure.ac	2015-02-11 18:46:39.048419499 +0100
@@ -13,7 +13,7 @@ AC_INIT(GStreamer Hantro G1 Decoder Plug
 AG_GST_INIT
 
 dnl initialize automake
-AM_INIT_AUTOMAKE([-Wno-portability 1.11 no-dist-gzip dist-xz tar-ustar])
+AM_INIT_AUTOMAKE([-Wno-portability 1.11 no-dist-gzip dist-xz tar-ustar foreign])
 
 dnl define PACKAGE_VERSION_* variables
 AS_VERSION
