--- qtbase.orig/src/plugins/platforms/linuxfb/qlinuxfbscreen.cpp	2014-08-30 18:51:54.497335792 -0400
+++ qtbase/src/plugins/platforms/linuxfb/qlinuxfbscreen.cpp	2014-08-30 18:52:36.125336669 -0400
@@ -181,6 +181,8 @@
             format = QImage::Format_RGB32;
             // pixeltype = BGRPixel;
         }
+	//Hack to force the format and alpha blending
+	format = QImage::Format_RGB32;
         break;
     }
     case 24: {
