From 47955d533df515b6ef3907b687cc6ad22c1430bf Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 16 Jan 2020 09:51:33 -0800
Subject: [PATCH] whiteboard.pro: Add qt to CONFIG in project

Fixes install errors when S != B

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 whiteboard.pro | 1 +
 1 file changed, 1 insertion(+)

--- a/whiteboard.pro
+++ b/whiteboard.pro
@@ -31,7 +31,7 @@ HEADERS  += mainwindow.h \
     whiteboard.h
 
 target.path = /opt/whiteboard
-target.files = whiteboard
+target.files = $$top_builddir/whiteboard
 extra.path = /opt/whiteboard
 extra.files = resources/whiteboard.sh
 configfile.path = /opt/ApplicationLauncher/applications/xml
