From f8da23d3f3fe8ae5119de5eee63ca567fbfc4838 Mon Sep 17 00:00:00 2001
From: Eugen Hristev <eugen.hristev@microchip.com>
Date: Mon, 24 Jun 2019 11:19:13 +0300
Subject: [PATCH] workaround to have video player work with QT 5.12

Signed-off-by: Eugen Hristev <eugen.hristev@microchip.com>
---
 player.cpp    |  4 +++-
 screen.config | 16 +++++++++-------
 2 files changed, 12 insertions(+), 8 deletions(-)

diff --git a/player.cpp b/player.cpp
index fe03f1a..cfe0fc5 100644
--- a/player.cpp
+++ b/player.cpp
@@ -73,7 +73,7 @@ Player::Player(QWidget *parent) :
     showFullScreen();
 
     // Set transparent background
-    setAttribute(Qt::WA_TranslucentBackground);
+//    setAttribute(Qt::WA_TranslucentBackground);
     setWindowFlags(Qt::FramelessWindowHint);
 
     centralWidget()->setMouseTracking(true);
@@ -82,6 +82,8 @@ Player::Player(QWidget *parent) :
     if (gem < 0)
         fprintf(stderr, "error: setup_planes() failed\n");
 
+    gem = 1; /* temporary hardcode the overlay to display the video */
+
     QRect rec = QApplication::desktop()->screenGeometry();
 
     // initialize variables
diff --git a/screen.config b/screen.config
index ec0c984..08da543 100644
--- a/screen.config
+++ b/screen.config
@@ -1,19 +1,21 @@
 {
     "planes": [
         {
-            "type": "primary",
-            "pattern": [ "0x000000ff" ],
-            "format": "DRM_FORMAT_RGBA8888"
+            "type": "overlay",
+            "index": 1,
+            "alpha": "0xFF",
+            "pattern": [ "0xFF000000" ],
+            "format": "DRM_FORMAT_XRGB8888"
         },
         {
-            "type": "overlay",
-            "index": 0,
+            "type": "primary",
+            "index": 1,
             "x": 0,
             "y": 0,
             "width": "SCREEN_WIDTH",
             "height": "SCREEN_HEIGHT",
-            "pattern": [ "0x00000000" ],
-            "format": "DRM_FORMAT_XRGB8888"
+            "pattern": [ "0xFF000000" ],
+            "format": "DRM_FORMAT_ARGB8888"
         }
     ]
 }
-- 
2.7.4

