From 2868ecf199a875ebdebf6d5601d24bab27f2b886 Mon Sep 17 00:00:00 2001
From: Eugen Hristev <eugen.hristev@microchip.com>
Date: Tue, 18 Jun 2019 11:37:31 +0300
Subject: [PATCH] fix PIPE semicolon

Signed-off-by: Eugen Hristev <eugen.hristev@microchip.com>
---
 videoplayer.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/videoplayer.cpp b/videoplayer.cpp
index 0f0631e..63af9e0 100644
--- a/videoplayer.cpp
+++ b/videoplayer.cpp
@@ -20,7 +20,7 @@
 
 #if defined PLANA || defined PLANC
 #define PIPE "uridecodebin expose-all-streams=false name=" SRC_NAME " \
-caps=video/x-h264;audio/x-raw " SRC_NAME ". ! queue ! h264parse ! \
+caps=video/x-h264\\;audio/x-raw " SRC_NAME ". ! queue ! h264parse ! \
 queue ! g1h264dec ! video/x-raw,width=%2,height=%3,format=BGRx ! \
 progressreport silent=true do-query=true update-freq=1 format=time \
 name=" PROGRESS_NAME " ! perf name=" PERF_NAME " ! \
@@ -29,7 +29,7 @@ audio/x-raw,format=S16LE ! volume name=" VOLUME_NAME " ! \
 alsasink async=false enable-last-sample=false"
 #else
 #define PIPE "uridecodebin expose-all-streams=false name=" SRC_NAME " \
-caps=video/x-h264;audio/x-raw " SRC_NAME ". ! queue ! h264parse ! \
+caps=video/x-h264\\;audio/x-raw " SRC_NAME ". ! queue ! h264parse ! \
 queue ! g1h264dec ! video/x-raw,width=%2,height=%3 ! \
 progressreport silent=true do-query=true update-freq=1 format=time \
 name=" PROGRESS_NAME " ! perf name=" PERF_NAME " ! \
-- 
2.7.4

